### PR TITLE
test: convert mock-adb to .NET

### PR DESCRIPTION
**Pull request by dbjorge**
#### Details

This PR migrates `mock-adb` from being a Node.js script packaged into an exe using `pkg` to being a .NET 6.0 console application built as a standalone exe.

##### Motivation

The main motivation is to improve the reliability of the unified tests by improving mock-adb performance. The .NET version is more than an order of magnitude faster than the Node.js version due to the high startup time involved to initialize the Node.js runtime, before our code starts running, and mock-adb is invoked tens of times per test for all unified tests of scenarios beyond the "device setup" flow.

##### Context

For internal folks, see the recording of our 7/7/22 engineering discussion for more context and team discussion. We considered and rejected a few other approaches, including:

* Rewrite in Rust/Go instead of .NET
* Move to a different repo as part of the rewrite
* Instead of rewriting, rearchitect some of the unified tests to avoid having to shell out to a separate `adb.exe` process so much

#### Pull request checklist

- [n/a] Addresses an existing issue: #0000
- [x] Ran `yarn fastpass`
- [no] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [n/a] (UI changes only) Added screenshots/GIFs to description above
- [n/a] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: packages/mock-adb/MockAdb.csproj [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+
+    <!--
+      These are required to produce a self-contained executable named "adb.exe", which is mandatory
+      because that's specifically what appium-adb looks for.
+    -->
+    <AssemblyName>adb</AssemblyName>
+    <OutputType>Exe</OutputType>
+	  <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
```
Comment (DaveTryon) [inline packages/mock-adb/MockAdb.csproj:15]: Indent seems off?
```suggestion
    <UseCurrentRuntimeIdentifier>true</UseCurrentRuntimeIdentifier>
```

File: packages/mock-adb/src/Program.cs [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+string currentExe = Assembly.GetExecutingAssembly().Location;
+string currentExeDir = Path.GetDirectoryName(currentExe) ?? throw new Exception("currentExe should not be a root directory");
+
+string defaultConfigPath = Path.Join(currentExeDir, "mock_adb_config.json");
+string configPath = Environment.GetEnvironmentVariable("MOCK_ADB_CONFIG") ?? defaultConfigPath;
+
+if (!File.Exists(configPath))
+{
+    Console.Error.WriteLine($"Could not find mock-adb config file at \"{configPath}\"");
+    Console.Error.WriteLine($"You can create one with \"yarn mock-adb\"");
```
Comment (DaveTryon) [inline packages/mock-adb/src/Program.cs:17]: Interpolated string adds no value on this line

File: src/tests/miscellaneous/mock-adb/README.md [hunk 3]
```diff
@@ -3,39 +3,8 @@ Copyright (c) Microsoft Corporation. All rights reserved.
 Licensed under the MIT License.
 -->
 
-## mock-adb
+# mock-adb test integration
 
-This directory contains the implementation for a `mock-adb` (Android Debug Bridge) executable. The unified product leverages `adb` to setup a test Android device. We use `mock-adb` in two ways:
-* during end-to-end tests that exercise the Android setup process
-* during development if you prefer to use a mock device/emulator
+This directory contains helpers to use mock-adb from within unified test cases.
```
Comment (jalkire) [inline src/tests/miscellaneous/mock-adb/README.md:8]: Suggestion: rename this directory to line up with this description (`src/tests/miscellaneous/mock-adb-helpers` for example). That way we have a more descriptive name and avoid having two `mock-adb` directories in different locations

File: packages/mock-adb/README.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+* `mock_adb_config.json` - the configuration used by `mock-adb` to control its behavior
+* `mock_adb_output.json` - the input and output for each invocation of `mock-adb`
+
+In the remote build, these logs are uploaded as artifacts if any end-to-end tests fail.
+
+### Implementation
+
+`mock-adb` itself is written in .NET for performance reasons; the Unified tests invoke it hundreds of times, and writing it in Node.js creates ~2s of overhead per call on some of the lower-end dev machines our team tests against.
+
+Commands such as `yarn mock-adb` and end-to-end tests call [setupMockAdb](https://github.com/microsoft/accessibility-insights-web/blob/main/src/tests/miscellaneous/mock-adb/setup-mock-adb.js#L28) to write the appropriate config/logging context to disk. `mock-adb` gets packaged into an executable that reads the context from disk and runs with the configured behavior.
```
Comment (jalkire) [inline packages/mock-adb/README.md:43]: This line number might be out of date--line 30 or no line number seem more appropriate to me
```suggestion
Commands such as `yarn mock-adb` and end-to-end tests call [setupMockAdb](https://github.com/microsoft/accessibility-insights-web/blob/main/src/tests/miscellaneous/mock-adb/setup-mock-adb.js) to write the appropriate config/logging context to disk. `mock-adb` gets packaged into an executable that reads the context from disk and runs with the configured behavior.
```

File: src/tests/miscellaneous/mock-adb/README.md (hunk 3 above)
Comment (dbjorge) [reply src/tests/miscellaneous/mock-adb/README.md]: Done! I went with `setup-mock-adb` instead of `mock-adb-helpers`

**Pull request merged**